doc/dev/transforms.rst

```diff
@@ -100,21 +98,154 @@ classical instructions as well as side-effecting operations (e.g., print) can al
 same representation.
 
 
+Writing and running your first Catalyst pass
+============================================
+
+.. note::
+
+    If you are encoutering issues, or would like to quickly try out the hello world pass described in this
+    section, you can have a look at or cherry-pick this commit which includes all changes described
+    in this section: https://github.com/PennyLaneAI/catalyst/commit/ba7b3438667963b307c07440acd6d7082f1960f3
+
+If this is your first time writing MLIR or LLVM passes, the boilerplate can be quite overwhelming. 
+Let's first set up the various boilerplate items required to register and run a new pass. 
+
+We'll create an empty pass in the ``Catalyst`` dialect that just prints out hello world to stdout.
+Note that the ``mlir/include`` (and ``mlir/lib``) directories consists of all the available dialects, so if you want to write a new pass in another dialect, it should be added to the subdirectory of that dialect. 
+
+The first thing to do is to create the pass object in the tablegen ``mlir/include/Catalyst/Transforms/Passes.td``:
```

Comment:
Can you add a reference to what is tablegen?

Reply:
done